pkgs/applications/version-management/got/default.nix

```diff
@@ -17,12 +17,18 @@ stdenv.mkDerivation rec {
   ++ lib.optionals stdenv.isDarwin [ libossp_uuid ];
 
   preConfigure = lib.optionals stdenv.isDarwin ''
-    # The configure script assumes dependencies on Darwin are install via
+    # The configure script assumes dependencies on Darwin are installed via
     # Homebrew or MacPorts and hardcodes assumptions about the paths of
     # dependencies which fails the nixpkgs configurePhase.
     substituteInPlace configure --replace 'xdarwin' 'xhomebrew'
   '';
 
+  patches = [ ./darwin.patch ];
+
+  preBuild = ''
+    buildFlagsArray+=(CFLAGS="-DHAVE_STRLCPY=1 -DHAVE_STRLCAT=1 -DHAVE_STRNSTR=1 -DHAVE_STRMODE=1")
+  '';
```

Comment:
Without these buildFlags the buildPhase fails on macOS with conflicting types for strmode.